### DISCUSSION TITLE
Can the route plan be updated through the API?

### DISCUSSION THREAD

**Discussion by ytipocmk631** [PedestrianDynamics/jupedsim#1404] [General]
I set up some queuing areas and connected to the waypoint behind the queue, but when the agent arrived at the waypoint, it did not move as expected (the agents route plan did not seem to be updated)

https://github.com/PedestrianDynamics/jupedsim/assets/129007821/f0291b80-b4bc-49bb-be6c-ba3eedc095d3

**Comment by chraibi** [comment]
What should happen when the waypoint is reached? Perhaps the agent needs to move to the exit, right?

Did you set a transition from the waypoint to the exit?
(`set_transition_for_stage`)

**Comment by Ozaq** [comment]
Could be bug in routing, can you please share the geometry? Either as python code or as WKT? This way I can have a look at it

Edit: Or if possible can you attach your python script to run your simulation so that we can have a look?

**Reply by ytipocmk631** [reply]
sure.
example in here: [example.zip](https://github.com/PedestrianDynamics/jupedsim/files/15160842/example.zip)

**Reply by Ozaq** [reply]
Thanks I will have a look into it

**Reply by Ozaq** [reply]
@ytipocmk631 It is a bug in our navigation, it looks severe and I have no idea how long it will take to address this. I have created an issue for it here: https://github.com/PedestrianDynamics/jupedsim/issues/1411
